internal/integration/infra_sdk_emitter.go

```diff
@@ -29,12 +29,33 @@ var removedAttributes = map[string]struct{}{
 
 // InfraSdkEmitter is the emitter using the infra sdk to output metrics to stdout
 type InfraSdkEmitter struct {
-	synthesisRules synthesis.Synthesizer
+	synthesisRules      synthesis.Synthesizer
+	integrationMetadata Metadata
+}
+
+// Metadata contains the name and version of the exporter that is being scraped.
+// The Infra-Agent use the metadata to populate instrumentation.name and instrumentation.value
+type Metadata struct {
+	Name    string `mapstructure:"name"`
+	Version string `mapstructure:"version"`
+}
+
+func (im *Metadata) isValid() bool {
+	return im.Name != "" && im.Version != ""
```

Comment:
should we log something in case one of the two is populated?

Reply:
We could perform this check when the metadata is being replaced and throw an error.